src/plot-data.ts

```diff
@@ -749,52 +749,101 @@ export abstract class PlotData {
         this.refresh_point_list_bool = true;
       }
       if (this.point_families.length == 0) {
-        for (var i=0; i<this.scatter_points.length; i++) {
-          var point:Point2D = this.scatter_points[i];
+        for (var i = 0; i < this.scatter_points.length; i++) {
+          var point: Point2D = this.scatter_points[i];
           this.draw_point(hidden, mvx, mvy, this.scaleX, this.scaleY, point);
         }
       } else {
         var point_order = this.get_point_order();
-        for (let i=0; i<point_order.length; i++) {
-          for (let j=0; j<point_order[i].length; j++) {
+        for (let i = 0; i < point_order.length; i++) {
+          for (let j = 0; j < point_order[i].length; j++) {
             let index = point_order[i][j];
-            let point:Point2D = this.scatter_points[index];
+            let point: Point2D = this.scatter_points[index];
             this.draw_point(hidden, mvx, mvy, this.scaleX, this.scaleY, point);
           }
         }
       }
 
-      for (var i=0; i<this.tooltip_list.length; i++) {
-        if (!List.is_include(this.tooltip_list[i],this.scatter_points)) {
+      for (var i = 0; i < this.tooltip_list.length; i++) {
+        if (!List.is_include(this.tooltip_list[i], this.scatter_points)) {
           this.tooltip_list = List.remove_element(this.tooltip_list[i], this.tooltip_list);
         }
       }
-      this.draw_tooltip(d.tooltip, mvx, mvy, this.scatter_points, d.point_list, d.elements, this.mergeON, 
+      this.draw_tooltip(d.tooltip, mvx, mvy, this.scatter_points, d.point_list, d.elements, this.mergeON,
+        d.attribute_names);
+    }
+  }
+
+
+  drawPiechart(d: PieChart, hidden: boolean, mvx: number, mvy: number): void {
```

Comment:
I suggest that you try to make this method clearer alone. As some comments I made elsewhere apply here too. 
Is it fine by you ?

Reply:
Tried to do something :)

Reply:
Moved and split in the PlotPieChart class